test/cmd/authentication.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+run_exec_credentials_tests() {
+  set -o nounset
+  set -o errexit
+
+  kube::log::status "Testing kubectl with configured exec credentials plugin"
+
+  cat > "${TMPDIR:-/tmp}"/invalid_exec_plugin.yaml << EOF
+apiVersion: v1
+clusters:
+- cluster:
+  name: test
+contexts:
+- context:
+    cluster: test
+    user: invalid_token_user
+  name: test
+current-context: test
+kind: Config
+preferences: {}
+users:
+- name: invalid_token_user
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1beta1
+      # Any invalid exec credential plugin will do to demonstrate
+      command: ls
+EOF
+
+  ### Provided --token should take precedence, thus not triggering the (invalid) exec credential plugin
+  # Pre-condition: None
+  # Command
+  output=$(kubectl "${kube_flags_with_token[@]:?}" --kubeconfig="${TMPDIR:-/tmp}"/invalid_exec_plugin.yaml get namespace kube-system -o name || true)
```

Comment:
Ah I just realized we need to run this command the other way as well:

```bash
output2=$(kubectl --kubeconfig="${TMPDIR:-/tmp}"/invalid_exec_plugin.yaml get namespace kube-system -o name || true)
```

Without the token it should fail with the parse error (which confirms that the `exec` plugin is at least attempting to run).

Reply:
Added! Needed to use the kube_flags still but without the token or else it would default to the insecure port.